langlib/langlib-test/src/test/resources/test-src/decimallib_test.bal

```diff
@@ -580,6 +580,40 @@ function testQuantizeFunctionWithInvalidOutput() {
     assertEquality("{ballerina/lang.decimal}QuantizeError", (<error> a1).message());
 }
 
+function testAvg() {
+    decimal a1 = decimal:avg(1d, 2d, 3d, -4d, -5d);
+    test:assertValueEqual(-0.6d, a1);
+
+    a1 = 0d.avg(0, 1);
+    test:assertValueEqual(0.3333333333333333333333333333333333d, a1);
+
+    a1 = decimal:avg(0, 1);
+    test:assertValueEqual(0.5d, a1);
+
+    a1 = decimal:avg(<decimal>int:MAX_VALUE, <decimal>int:MAX_VALUE, <decimal>int:MAX_VALUE, <decimal>int:MAX_VALUE);
+    test:assertValueEqual(<decimal>int:MAX_VALUE, a1);
+
+    a1 = int:avg(int:MAX_VALUE, int:MIN_VALUE);
+    test:assertValueEqual(-0.5d, a1);
+
+    decimal[] arr = [2, 3, 4, 5];
+    a1 = decimal:avg(1, ...arr);
+    test:assertValueEqual(3d, a1);
+
+    a1 = decimal:avg(1, ...[2, 3, 4, 5]);
+    test:assertValueEqual(3d, a1);
+
+    arr = [];
+    foreach int i in 0...1000 {
+        arr.push(<decimal>int:MAX_VALUE);
+    }
+    test:assertValueEqual(<decimal>int:MAX_VALUE, decimal:avg(<decimal>int:MAX_VALUE, ...arr));
+
+    decimal|error d = trap decimal:avg(9.999999999999999999999999999999999e6144d, 9.999999999999999999999999999999999e6144d);
+    test:assertValueEqual(true, d is error);
+    test:assertValueEqual("{ballerina}NumberOverflow", (<error>d).message());
```

Comment:
@hasithaa, when calculating the avg, first we have to get some. Therefore for long:avg function, there is a possibility to have an overflow error. This is expected, right?

Reply:
Can we see how other implementations handle this?

Reply:
@hasithaa, I check this is Java and SQL in the MySql server and python.

1. In Java there is not an inbuilt avg function for BigDecimal (equivalent to ballerina decimal). But there is a function called `IntStream.average()` which returns `OptionalDouble`. There is no possibility to have a number overflow in this function.
2. In SQL, If we get the average of Double max, using its' average function the answer is `0`.
3. In python, it gives an overflow error for the following example
```python
import numpy
def Average(lst):
  avg = numpy.average(lst)
  return(avg)
lst = [9.999999999999999999999999999999999e307, 9.999999999999999999999999999999999e307]
print("Average of the list =", Average(last))
```

Reply:
Ok, Can we implement something like IntStream? overflow on avg function doesn't seems right.

Reply:
`int:avg` does not overflow because it returns `decimal`. This is the same as IntStream which returns Double. 
But in our case, overflow happens to `decimal:avg`, it returns a decimal.